### ISSUE TITLE
for loops and let do not report errors correctly

### ISSUE THREAD

**Issue by arv** [google/traceur-compiler#1820]
I had:

```js
// Options: --script

function f() {
  'use strict';
  for (let iii = 0; iii < 5; iii++) {}
  for (iii = 0; iii < 5; iii++) {}
}
f();
```

This should be an error but the code we generate is:

```js
function f() {
  'use strict';
  for (var iii = 0; iii < 5; iii++) {}
  for (iii = 0; iii < 5; iii++) {}
}
f();
```

which is clearly wrong.
